capplets/time-admin/src/time-zone.c

```diff
@@ -269,17 +262,22 @@ static void
 update_timezone (TimezoneMap *map)
 {
     g_autofree gchar *bubble_text = NULL;
+    g_autoptr(GDateTime) current_date = NULL;
+    g_autoptr(GTimeZone) timezone = NULL;
     g_autofree gchar *city_country = NULL;
     g_autofree gchar *utc_label = NULL;
     g_autofree gchar *time_label = NULL;
     g_autofree gchar *tz_desc = NULL;
     TzLocation       *current_location;
     GDateTime        *date;
 
-    date = g_date_time_new_now_local ();
     current_location = timezone_map_get_location (TIMEZONEMAP (map));
     city_country = translated_city_name (current_location);
     
+    timezone = g_time_zone_new (current_location->zone);
+    current_date = g_date_time_new_now_local ();
+    date = g_date_time_to_timezone (current_date, timezone);
```

Comment:
You should release the return value of[ g_date_time_to_timezone()](https://developer.gnome.org/glib/stable/glib-GDateTime.html#g-date-time-to-timezone) by calling [g_date_time_unref()](https://developer.gnome.org/glib/stable/glib-GDateTime.html#g-date-time-unref) when you are done with it.

Reply:
It's my oversight. I've changed the problem.